src/babel-plugin-import-meta.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import importMetaSyntax from '@babel/plugin-syntax-import-meta';
+import template from '@babel/template';
+import {NodePath} from '@babel/traverse';
+import {MetaProperty} from '@babel/types';
+import {relative} from 'path';
+
+const ast = template.ast;
+
+/**
+ * Rewrites `import.meta`[1] into an object with a `url`[2] property.
+ *
+ * `import.meta.url` must be a URL string with the fully qualified URL of the
+ * module. We use the document's base URI and the relative path from rootDir to
+ * filePath to build the URL.
+ *
+ * [1]: https://github.com/tc39/proposal-import-meta
+ * [2]: https://html.spec.whatwg.org/#hostgetimportmetaproperties
+ *
+ * @param filePath THe path of the file being transformed
+ * @param rootDir The root project folder containing filePath
+ * @param base A base URL to use instead of document.baseURI
+ */
+export const rewriteImportMeta = (
+    filePath: string,
+    rootDir: string,
+    base?: string,
+    ) => {
+  return {
+    inherits: importMetaSyntax,
+    visitor: {
+      MetaProperty(path: NodePath<MetaProperty>) {
+        const node = path.node;
+        if (node.meta && node.meta.name === 'import' &&
+            node.property.name === 'meta') {
+          const relativePath = relative(rootDir, filePath);
```

Comment:
If either `rootDir` or `filePath` are relative, `path.relative` will first evaluate them as relative to `cwd`, and then compute a relative path from one to the other. I feel like this could be a little confusing.

Maybe another approach would be to just take the final relative URL, to shift the burden of computing a relative path (if needed) to caller?

Also on windows `relative` will return a backslash delimited path, so that would need to be fixed either way I think.

Reply:
> Maybe another approach would be to just take the final relative URL, to shift the burden of computing a relative path (if needed) to caller?

I waved on this myself. I thought that if the computation of the final relative path was always the same, then we could safer handle it here. Your points swayed me though. It's also possible that a caller just already has the relative path.

With this change I think it should be a relative URL though, and not do any Windows handling here.